### DISCUSSION TITLE
M2 showing up for first time, but unavailable

### DISCUSSION THREAD

**Discussion by jpravetz** [WillCodeForCats/solaredge-modbus-multi#410] [General]
Slowly my SolarEdge monitoring is getting better, and I am overcoming [Known Issues](https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Known-Issues). First my LG Battery magically started reporting, which it did not initially do. What I'm still lacking, however, is the measurement for my own power consumption.

Today I see that SolarEdge M2 is showing up and I am seeing all 4 devices (yay!). But I get the following error on startup. 

```
Logger: custom_components.solaredge_modbus_multi.hub
Source: custom_components/solaredge_modbus_multi/hub.py:389
Integration: SolarEdge Modbus Multi ([documentation](https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki), [issues](https://github.com/WillCodeForCats/solaredge-modbus-multi/issues))
First occurred: 5:18:46 PM (1 occurrences)
Last logged: 5:18:46 PM
Duplicate serial 606969163 on meter 2 inverter 1
```

and  all the fields for the M2 are _unavailable_.

The M1 device
```
SE-RGMTR-1D-240C-B
by SolarEdge
Connected via [Solaredge I1]
Firmware: 78
```

The M2 device
```
SE-RGMTR-1D-240C-B
by SolarEdge
Connected via [Solaredge I1]
Firmware: 78
```

And I1 says Production.

I had been communicating with SolarEdge support, because of not being able to see the 2nd meter for consumption, and today's response to our continued thread is

_Please keep in mind that Modbus is a form of external meter that is used to read consumption, self consumption, and production on the system, whereas the E/I function reads power export and import from the grid. Your system looks to be already set up to do E/I functions. The inverter's firmware has already been upgraded. If you want to check the I/E status of the inverter, you can check this on SetApp and look for "Status" to see the current Import and Export since you mentioned that you have an access to it._

I had told them I could see import/export, but not consumption. So I'm a bit confused about whether they are answering my initial question about getting access to my two meter's data.

Any help from @WillCodeForCats or others?

**Comment by WillCodeForCats** [comment]
Do you actually have two physical, external meters?

Or do you only have one meter configured for dual function?

**Comment by jpravetz** [comment]
No, not two external meters.
I _assume_ I have one meter that presents itself as two meters, as per your
earlier discussions, such as under 'Known Issues'.

thanks, jim

**Reply by WillCodeForCats** [reply]
So if that's the case and SolarEdge has indeed fixed the previous issue where the two-in-one meter situation does not report the second function, they would probably report the same serial number and there is a duplicate serial number check in the setup sequence.

In that case I can probably safely remove the duplicate serial number check, after a make sure we aren't still using the serial number for anything breaking.

**Reply by WillCodeForCats** [reply]
PR #412 will remove the duplicate serial number check for meters (but still log the warning)

**Comment by WillCodeForCats** [comment]
Try installing [Release v2.4.3-pre.6](https://github.com/WillCodeForCats/solaredge-modbus-multi/releases/tag/v2.4.3-pre.6)

This release removes the duplicate meter serial number check.

**Comment by jpravetz** [comment]
The fix seems to work, thanks! Here is [an expiring link to screenshots of my i1, b1, m1, m2](https://www.dropbox.com/scl/fi/oak8fl3jdeqqaf635wiwn/solareedge_modbus.zip?rlkey=dvb0252wiafr496oce8rqgmfi&dl=0), in case you wish to review them. 

Thank you again for your help!

**Reply by WillCodeForCats** [reply]
Can you upload your diagnostic file? It contains additional info that screenshots don't have.

**Comment by jpravetz** [comment]
[diagnostic file](https://www.dropbox.com/scl/fi/631lwmy5zo0fftmgmcxec/config_entry-solaredge_modbus_multi-4848c32ab2f3d070ad779583f2ee4267.json.txt?rlkey=tmwakl8jml16llrb3uwucjeyi&dl=0)

**Comment by jpravetz** [comment]
I'm seeing now that the measurement numbers for M1 and M2 are identical.

**Reply by WillCodeForCats** [reply]
Yes, that's what I see in the diagnostic file.

I don't have a problem with letting the integration report two meters like this, but I don't see how it's useful.

**Comment by jpravetz** [comment]
Certainly if it shows the same data it's not useful. I thought the idea was that the 2nd meter would show the measurements off a CT clamped to my consumption. I'm missing my consumption number, or I'm just not interpreting the data fields correctly.

**Reply by WillCodeForCats** [reply]
If you have multiple physical meters they can each have a defined function: Production, Consumption, or Export/Import. There's also Site Production and External Production options for advanced configurations.

The (known) problem comes into play with a special option to use a single physical meter for multiple functions: Production + Consumption or Production + Export/Import. This is found on inverters that ship with a single revenue grade meter built in. SolarEdge splits CT1/CT2 from CT3 in software to use that single meter in a way that would normally take two physical meters. Unfortunately this use case does not work correctly over the modbus interface.

If you have separate, single function physical meters they will report over modbus correctly.

This document describes the dual function meter: https://knowledge-center.solaredge.com/sites/kc/files/se-inverters-with-rgm-and-export-import-meter-na.pdf

**Comment by jpravetz** [comment]
Thanks for your help, Seth. I will look into it.

**Comment by jpravetz** [comment]
SolarEdge responded 

_Thank you for clarifying what you need. I was able to determine the issue on your monitoring and verified that the consumption and self consumption was disabled. Configured on my end and enable both meters. Please try to observe for at least 24 to 48 hours before the data will reflect on your end._

Now I wait to see what this does.

**Reply by WillCodeForCats** [reply]
You will want to also reload the integration to re-interview the meters.

**Reply by Tcoton** [reply]
2 years later, did you get it working? I also have a Production+ Import/export meter (SE-RGMTR-1D-240C-C) and it drives me crazy not to be able to read the CT data while I can see them when connected via setApp or the internal website of the inverter.

**Reply by WillCodeForCats** [reply]
You might need a firmware update on the inverter: https://github.com/WillCodeForCats/solaredge-modbus-multi/discussions/298#discussioncomment-14776200

**Reply by Tcoton** [reply]
Firmware update requested, let's see how it goes!
